src/util/fresh_symbol.cpp

```diff
@@ -38,11 +38,10 @@ symbolt &get_fresh_aux_symbol(
   const irep_idt &symbol_mode,
   symbol_tablet &symbol_table)
 {
-  auxiliary_symbolt new_symbol;
-
   // Loop until find a name that doesn't clash with a non-auxilliary symbol
   while(true)
   {
+    auxiliary_symbolt new_symbol;
```

Comment:
There seems to be further room for code beautification:
````c++
auxiliary_symbolt new_symbol(
  basename_prefix+"$"+std::to_string(++temporary_counter),
  type);
if(!name_prefix.empty())
  new_symbol.name=name_prefix+"::"+id2string(new_symbol.base_name);
````
while getting rid of all the lines up to the `.type` assignment; but maybe that's only only me seeing this as more beautiful.

Reply:
That change seems within scope of this change. Applied.